src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -1766,6 +1787,49 @@ FindAndMoveShardCost(float4 utilizationLowerBound, float4 utilizationUpperBound,
 					 */
 					continue;
 				}
+
+				/*
+				 * fmaxf and fminf here are only needed for cases when nodes
+				 * have different capacities. If they are the same, then both
+				 * arguments are equal.
+				 */
+				float4 utilizationImprovement = fmaxf(
+					sourceFillState->utilization - newTargetUtilization,
+					newSourceUtilization - targetFillState->utilization
+					);
+				float4 utilizationAddedByShard = fminf(
+					newTargetUtilization - targetFillState->utilization,
+					sourceFillState->utilization - newSourceUtilization
+					);
+
+				/*
+				 * If the shard causes a lot of utilization, but the
+				 * improvement is gained by moving it is small then we ignore
+				 * it. Probably there are other shards that are better
+				 * candidates, and in any case it's probably not worth the
+				 * effort to move the this shard.
```

Comment:
This paragraph is a bit hard to understand, is it supposed to be like below? If not please reword yourself to make it reflect the reality better.

```suggestion
				 * If the shard causes a lot of utilization, but the
				 * improvement which is gained by moving the shard is small then we ignore
				 * the move. Probably there are other shards that are better
				 * candidates, and in any case it's probably not worth the
				 * effort to move the this shard.
```